### PR TITLE
 homematic add-on to use Bashio #725

### LINKED ISSUE #725: Update homematic add-on to use Bashio
The add-ons now have [Bashio](https://github.com/hassio-addons/bashio) available, which makes it easier and nicer to work with and is easier to understand. The `homematic` add-on needs to be rewritten to use the Bashio functions.

In general, this only requires changes to the add-on [`run.sh`](https://github.com/home-assistant/hassio-addons/blob/master/homematic/data/run.sh) script. A good example of an add-on that has been migrated already, is [`deconz`](https://github.com/home-assistant/hassio-addons/blob/master/deconz/data/run.sh).

Please note, nothing has to be done to make Bashio available, it is already baked-in into the add-on base images.

### PR DESCRIPTION
**Pull request by Zapfmeister**
Replace bash with bashio.
Replace JQ with bashio.

Closes #725

### CODE REVIEW

File: homematic/data/run.sh [hunk 1]
```diff
@@ -37,11 +35,11 @@ init_interface_list "$RF_ENABLE" "$HMIP_ENABLE" "$WIRED_ENABLE"
 # RF support
 if [ "$RF_ENABLE" == "true" ]; then
```
Comment (frenck) [inline homematic/data/run.sh]: Bashio can check this directly.

```suggestion
if bashio::config.true 'rf_enable'; then
```

File: homematic/data/run.sh [hunk 2]
```diff
@@ -37,11 +35,11 @@ init_interface_list "$RF_ENABLE" "$HMIP_ENABLE" "$WIRED_ENABLE"
...
     for (( i=0; i < "$RF_DEVICES"; i++ )); do
```
Comment (frenck) [inline homematic/data/run.sh]: This for loop can be changed to a loop over the keys. 

See example here: <https://github.com/hassio-addons/addon-ftp/blob/master/ftp/rootfs/etc/cont-init.d/users.sh#L9>

This eliminates the use of a length based variable as well.

File: homematic/data/run.sh [hunk 3]
```diff
@@ -76,9 +74,9 @@ fi
 # Wired support
 if [ "$WIRED_ENABLE" == "true" ]; then
```
Comment (frenck) [inline homematic/data/run.sh]: Use `bashio::config.true "config_var_name"`

File: homematic/data/run.sh [hunk 4]
```diff
@@ -76,9 +74,9 @@ fi
...
     for (( i=0; i < "$WIRED_DEVICES"; i++ )); do
```
Comment (frenck) [inline homematic/data/run.sh]: Change the loop by iterating over keys as mentioned above.

File: homematic/data/run.sh [hunk 5]
```diff
@@ -107,8 +105,8 @@ if [ "$HMIP_ENABLE" == "true" ]; then
 
     # Setup settings
     for (( i=0; i < "$HMIP_DEVICES"; i++ )); do
```
Comment (frenck) [inline homematic/data/run.sh]: Change the loop by iterating over keys as mentioned above.

File: homematic/data/run.sh [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
 
-CONFIG_PATH=/data/options.json
-
-RF_ENABLE=$(jq --raw-output '.rf_enable' $CONFIG_PATH)
-RF_DEVICES=$(jq --raw-output '.rf | length' $CONFIG_PATH)
-WIRED_ENABLE=$(jq --raw-output '.wired_enable' $CONFIG_PATH)
-WIRED_DEVICES=$(jq --raw-output '.wired | length' $CONFIG_PATH)
-HMIP_ENABLE=$(jq --raw-output '.hmip_enable' $CONFIG_PATH)
-HMIP_DEVICES=$(jq --raw-output '.hmip | length' $CONFIG_PATH)
+RF_ENABLE=$(bashio::config 'rf_enable')
```
Comment (frenck) [inline homematic/data/run.sh]: None of these 6 variables are needed anymore and can be replaced in line with Bashio.
See the other comments I made on this PR.

File: homematic/data/run.sh [hunk 7]
```diff
@@ -74,11 +67,11 @@ if [ "$RF_ENABLE" == "true" ]; then
 fi
 
 # Wired support
-if [ "$WIRED_ENABLE" == "true" ]; then
-    for (( i=0; i < "$WIRED_DEVICES"; i++ )); do
-        SERIAL=$(jq --raw-output ".wired[$i].serial" $CONFIG_PATH)
-        KEY=$(jq --raw-output ".wired[$i].key" $CONFIG_PATH)
-        IP=$(jq --raw-output ".wired[$i].ip" $CONFIG_PATH)
+if bashio::config.true 'WIRED_ENABLE'; then
```
Comment (frenck) [inline homematic/data/run.sh]: There is no configuration option `WIRED_ENABLE` in the configuration schema

File: homematic/data/run.sh [hunk 8]
```diff
@@ -32,16 +25,16 @@ touch /data/homematic.regadom
 . /usr/lib/hm-interface.sh
 
 # Setup Interfaces
-init_interface_list "$RF_ENABLE" "$HMIP_ENABLE" "$WIRED_ENABLE"
+init_interface_list "$(bashio::config 'rf_enable')" "$(bashio::config 'hmip_enable')" "$(bashio::config 'wired_enable')"
```
Comment (frenck) [inline homematic/data/run.sh]: Lets make this line shorter for readability 
```suggestion
init_interface_list \
    "$(bashio::config 'rf_enable')" \
    "$(bashio::config 'hmip_enable')" \
    "$(bashio::config 'wired_enable')"
```

File: homematic/data/run.sh [hunk 9]
```diff
@@ -1,15 +1,8 @@
...
 
-CONFIG_PATH=/data/options.json
-
-RF_ENABLE=$(jq --raw-output '.rf_enable' $CONFIG_PATH)
-RF_DEVICES=$(jq --raw-output '.rf | length' $CONFIG_PATH)
-WIRED_ENABLE=$(jq --raw-output '.wired_enable' $CONFIG_PATH)
-WIRED_DEVICES=$(jq --raw-output '.wired | length' $CONFIG_PATH)
-HMIP_ENABLE=$(jq --raw-output '.hmip_enable' $CONFIG_PATH)
-HMIP_DEVICES=$(jq --raw-output '.hmip | length' $CONFIG_PATH)
+RF_ENABLE=$(bashio::config 'rf_enable')
```
[the remark this answers is not in the document]
Comment (Zapfmeister) [reply homematic/data/run.sh]: Not after removing the last loop where i missed to replace it with bashio. Done now

**Comment by frenck** [comment]
In general, it looks good to me now (based on pure a visual review). Please update the changelog, and the bump the add-on version in the `config.json`.

**Comment by pvizeli** [comment]
Can you also update the log schema?

**Comment by Zapfmeister** [comment]
Can you please review the pull request. i guess it should be fine to be merged?

**Pull request merged**